src/python/bcc/__init__.py

```diff
@@ -388,7 +388,6 @@ def attach_kprobe(self, event="", fn_name="", event_re="",
         return self
 
     def detach_kprobe(self, event):
-        assert isinstance(event, str), "event must be a string"
```

Comment:
Was this being hit in practice? If str() is being wrapped around all internal decode() uses, wouldn't str be the default option?